### PR TITLE
refactor: Rename `StringBone`'s `maxLength` into `max_length`

### PR DESCRIPTION
**Pull request by ArneGudermann**
Replace all `maxLength` with `max_length` and a backwards compatible check with a warning.

### CODE REVIEW

File: src/viur/core/render/json/default.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def render_structure(structure: dict):
                     "boundslng": "boundsLng",
                     "emptyvalue": "emptyValue",
                     "max": "maxAmount",
-                    "maxlength": "maxLength",
+                    "max_length": "max_length",
```
Comment (sveneberth) [inline src/viur/core/render/json/default.py:50]:
```suggestion
                    "maxlength": "max_length",
```

File: src/viur/core/utils.py [hunk 2]
```diff
@@ -67,14 +69,19 @@ def markFileForDeletion(dlkey: str) -> None:
...
         Quotes several characters and removes "\\\\n" and "\\\\0" to prevent XSS injection.
 
         :param val: The value to be escaped.
-        :param maxLength: Cut-off after maxLength characters. A value of 0 means "unlimited".
+        :param max_length: Cut-off after max_length characters. A value of 0 means "unlimited".
         :returns: The quoted string.
     """
+    # fixme: Remove in viur-core >= 4
+    if "maxLength" in kwargs:
+        warnings.warn("maxLength is deprecated, please use max_length")
```
Comment (sveneberth) [inline src/viur/core/utils.py:82]: Sorry, my suggestion in the other PR wasn't complete, please add the class:
```suggestion
        warnings.warn("maxLength parameter is deprecated, please use max_length", DeprecationWarning)
```

File: src/viur/core/bones/text.py [hunk 3]
```diff
@@ -349,27 +350,31 @@ def __init__(
...
-            :param maxLength: Limit content to maxLength bytes
-            :param indexed: Must not be set True, unless you limit maxLength accordingly
+            :param max_length: Limit content to max_length bytes
+            :param indexed: Must not be set True, unless you limit max_length accordingly
             :param srcSet: If set, inject srcset tags to embedded images. Must be a dict of
                 "width": [List of Ints], "height": [List of Ints], eg {"height": [720, 1080]}
         """
+        # fixme: Remove in viur-core >= 4
+        if "maxLength" in kwargs:
+            warnings.warn("maxLength is deprecated, please use max_length")
```
Comment (sveneberth) [inline src/viur/core/bones/text.py:368]: Sorry, my suggestion in the other PR wasn't complete, please add the class:
```suggestion
            warnings.warn("maxLength parameter is deprecated, please use max_length", DeprecationWarning)
```

File: src/viur/core/bones/string.py [hunk 4]
```diff
@@ -302,6 +308,6 @@ def getUniquePropertyIndexValues(self, skel, name: str) -> List[str]:
 
     def structure(self) -> dict:
         ret = super().structure() | {
-            "maxlength": self.maxLength
+            "max_length": self.max_length
```
Comment (sveneberth) [inline src/viur/core/bones/string.py:311]:
```suggestion
            "maxlength": self.max_length
```

File: src/viur/core/bones/string.py [hunk 5]
```diff
@@ -32,11 +34,15 @@ def __init__(
             that creates the value for the index property.
         :param kwargs: Inherited arguments from the BaseBone.
         """
+        # fixme: Remove in viur-core >= 4
+        if "maxLength" in kwargs:
+            warnings.warn("maxLength is deprecated, please use max_length")
```
Comment (sveneberth) [inline src/viur/core/bones/string.py:39]:
```suggestion
            warnings.warn("maxLength parameter is deprecated, please use max_length", DeprecationWarning)
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: src/viur/core/render/json/default.py [hunk 6]
```diff
@@ -47,7 +47,7 @@ def render_structure(structure: dict):
                     "boundslng": "boundsLng",
                     "emptyvalue": "emptyValue",
                     "max": "maxAmount",
-                    "maxlength": "maxLength",
+                    "maxlength": "max_length",
```
Comment (phorward) [inline src/viur/core/render/json/default.py:50]: This is invalid, as this is the compatibility for the old bone structure rendering, and there, "maxLength" is the old value required.
```suggestion
                    "maxlength": "maxLength",
```

File: src/viur/core/utils.py [hunk 7]
```diff
@@ -86,8 +93,8 @@ def escapeString(val: str, maxLength: int = 254) -> str:
         .replace("\n", "") \
         .replace("\0", "")
 
-    if maxLength:
-        return val[0:maxLength]
+    if max_length:
+        return val[0:max_length]
```
Comment (phorward) [inline src/viur/core/utils.py:97]:
```suggestion
        return val[:max_length]
```

**Review by phorward — CHANGES_REQUESTED** [review]
Hello @ArneGudermann, thanks for enforcing this, can you please correct the two suggestions?

**Comment by phorward** [comment]
@ArneGudermann can you please also resolve the conflicts which came in by #940 ?

**Review by sveneberth — APPROVED** [review]

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: src/viur/core/bones/string.py [hunk 8]
```diff
@@ -24,7 +26,8 @@ def __init__(
         Initializes a new StringBone.
 
         :param caseSensitive: When filtering for values in this bone, should it be case-sensitive?
-        :param maxLength: The maximum length allowed for values of this bone. Set to None for no limitation.
+
```
Comment (sveneberth) [inline src/viur/core/bones/string.py:29]:
```suggestion
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by sveneberth — APPROVED** [review]

**Review by phorward — APPROVED** [review]

**Pull request merged**